irc/client.go

```diff
@@ -108,6 +110,9 @@ func NewClient(server *Server, conn net.Conn, isTLS bool) *Client {
 	}
 	client.languages = server.languages.Default()
 
+	// Check IP towards speified DNSBLs
+	server.ProcessBlacklist(client)
```

Comment:
This is annoying: the client's IP can "change" due to gateways (e.g., WEBIRC), so I think you need to call this a second time in `ApplyProxiedIP` (same as with bans).